specutils/utils/spectral_region.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import astropy.units as u
+
+
+class SpectralRegion:
+    """
+    SpectralRegion is currently a container class enables some simplicty
+    in defining and passing a region (interval) for a spectrum.
+
+    In the future, there might be more functionality added in here and there
+    is some discussion that this might/could move to astropy.region.
+    """
+
+    def __init__(self, lower, upper):
+        """
+        Lower and upper values for the interval.
+
+        Parameter
+        --------
+
+        lower: number with ``astropy.units`` unit
+           The lower bound of the region.
+
+        upper: number with ``astropy.units`` unit
+           The upper bound of the region.
+
+        Notes
+        -----
+
+        The ``astropy.units`` for each do not have to be the same, but to retain
+        sanity it might be easiest to have them the same.  The will need to be
+        dispersion units as opposed to flux units.
+        """
+
+        if not isinstance(lower, u.Quantity):
+            raise TypeError('Lower bound of the region must have an astropy.units unit')
+
+        if not isinstance(upper, u.Quantity):
+            raise TypeError('Upper bound of the region must have an astropy.units unit')
+
+        if upper < lower:
+            raise ValueError('Lower bound, {}, must be less than the upper bound {}'.format(lower, upper))
+
+        self._lower = lower
+        self._upper = upper
+
+    @property
+    def lower(self):
+        return self._lower
+
+    @lower.setter
+    def lower(self, value):
+        """
+        Lower bound for the interval.
+
+        Parameter
+        --------
+
+        value: number with ``astropy.units`` unit
+           The lower bound of the region.
+        """
+
+        if not isinstance(value, u.Quantity):
+            raise TypeError('Lower bound of the region must have an astropy.units unit')
```

Comment:
You also want to check that the units are valid units.  E.g., they must be either `pixel`, or *if* the `spectral_axis` has physical units, it has to be something that can be converted to that using the `spectral` equivalency.  I think the easiest way to do this is:
```
if not (value.unit.is_equivalent(u.pixel)  or
           value.unit.is_equivalent(u.angstrom, equivalencies=u.spectral)):
    raise u.UnitError('Not a spectral region unit')
```
You could potentially also drop the type error with this, because if a user passes in a non-unit object, it will still raise an error that says ``does not have unit``, basically.  Arguably that's better because it's duck-typing, so a quantity-like thing that's not literally a quantity will still work... but your call which you think is safer.

Reply:
changed and will be in next merge